wsi_deid/process.py

```diff
@@ -312,6 +339,13 @@ def redact_item(item, tempdir):
             macroImage = redact_topleft_square(macroImage)
         except Exception:
             pass
+    macro_geojson = redactList.get('images', {}).get('macro', {}).get('geojson')
+    if macroImage is None and macro_geojson:
+        try:
+            macroImage = PIL.Image.open(io.BytesIO(tileSource.getAssociatedImage('macro')[0]))
+            macroImage = redact_image_area(macroImage, macro_geojson)
```

Comment:
In looking at this, and the similar redact_topleft_square code above, I think it would be better if both of these only guarded the initial load, so if the redaction fails we throw the error.  That is:
```
try:
    macroImage = PIL.Image.open(io.BytesIO(tileSource.getAssociatedImage('macro')[0]))
except Exception:
    pass
if macroImage:
    macroImage = redact_image_area(macroImage, macro_geojson)
```

Reply:
I did some refactoring here in 195b13b